src/main/java/gregtech/common/metatileentities/storage/CachedRecipeData.java

```diff
@@ -1,97 +1,56 @@
-            return false;
-        }
-        if (!consumeRecipeItems(false)) {
-            this.lastTickChecked = -1L;
-            return false;
-        }
-        ForgeHooks.setCraftingPlayer(player);
-        InventoryCrafting deepCopy = InventoryUtils.deepCopyInventoryCrafting(inventory);
-        NonNullList<ItemStack> remainingItems = recipe.getRemainingItems(deepCopy); // todo right here is where tools get damaged (in UI)
-        ForgeHooks.setCraftingPlayer(null);
-        for (ItemStack itemStack : remainingItems) {
-            itemStack = itemStack.copy();
-            ItemStackKey stackKey = new ItemStackKey(itemStack);
-            int remainingAmount = itemStack.getCount() - itemSourceList.insertItem(stackKey, itemStack.getCount(), false, InsertMode.HIGHEST_PRIORITY);
-            if (remainingAmount > 0) {
-                itemStack.setCount(remainingAmount);
-                player.addItemStackToInventory(itemStack);
-                if (itemStack.getCount() > 0) {
-                    player.dropItem(itemStack, false, false);
-                }
-            }
-        }
-        this.lastTickChecked = -1L;
-        return true;
+        this.inventory = inventoryCrafting;
     }
 
     public short attemptMatchRecipe() {
-        ingredientsMatched = true;
         short itemsFound = 0;
+        if (recipe == null) {
+            return 511;
+        }
         this.requiredItems.clear();
+        boolean allItemsFound = true;
         for (int i = 0; i < inventory.getSizeInventory(); i++) {
             if (getIngredientEquivalent(i))
                 itemsFound += 1 << i; //ingredient was found, and indicate in the short of this fact
-            else
-                ingredientsMatched = false;
+            else {
+                allItemsFound = false;
+            }
+        }
+        if (!allItemsFound) {
+            requiredItems.clear();
         }
         return itemsFound;
     }
 
-    public boolean checkRecipeValid() {
-        if (!ingredientsMatched) {
+    protected boolean consumeRecipeItems() {
+        if (requiredItems.isEmpty()) {
             return false;
         }
-        long currentTick = itemSourceList.getWorld().getTotalWorldTime();
-        //check only once a tick if multiple players are viewing crafting table inventory
-        if (this.lastTickChecked == currentTick) {
-            return recipeValidCache;
-        }
-        this.lastTickChecked = currentTick;
-        this.recipeValidCache = consumeRecipeItems(true);
-        return recipeValidCache;
-    }
-
-    private boolean consumeRecipeItems(boolean simulate) {
         for (Entry<ItemStackKey, Integer> entry : requiredItems.entrySet()) {
             ItemStackKey itemStackKey = entry.getKey();
-            if (itemSourceList.extractItem(itemStackKey, entry.getValue(), simulate) != entry.getValue()) {
+            if (itemSources.extractItem(itemStackKey, entry.getValue(), false) != entry.getValue()) {
```

Comment:
```
if(itemSources.extractItem(simulate = true) == entry.getValue) {
    itemSources.extractItem(simulate = false)
}
else {
    return false;
}
```

Simulation check first?

Reply:
simulated extraction check is already performed by getIngredientEquivalent